Include/internal/pycore_pymem.h

```diff
@@ -100,6 +100,18 @@ static inline void _PyObject_XDecRefDelayed(PyObject *obj)
 }
 #endif
 
+#ifdef Py_GIL_DISABLED
+// This is the delayed-free equivalent of Py_XSETREF(), providing the same
+// atomic reference assignment semantics but with deferred cleanup suitable
+// for concurrent access patterns in free-threaded Python.
```

Comment:
```suggestion
// Same as `Py_XSETREF` but in free-threading, it stores the object atomically
// and queues the old object to be decrefed at a safe point using QSBR
```

Consider something like it